components/config-mgmt-service/integration_test/rollout_progress_test.go

```diff
@@ -0,0 +1,517 @@
+package integration_test
+
+/*
+
+TODO LIST:
+* fix the mark missing so we retain the prior status in the database
+* make sure crap like trailing `/` on URLs, http vs. https doesn't affect matching of rollouts
+* Use "errored" instead of failed to describe CCRs that had an error
```

Comment:
Also: add APIs to detail the known node segments (at least) so users can troubleshoot if their rollouts and CCRs aren't matching up.